[feat] 댓글 등록 시 comment_count 기능

yuhandemian opened the pull request:
## Issues
- closed #75
## ✔️  Check-list
- [x] : Label을 지정해 주세요.
- [x] : Merge할 브랜치를 확인해 주세요.

## 🗒️ Work Description

댓글 생성/삭제 시 Article의 commentCount가 자동으로 업데이트되도록 구현

### 변경 사항
1. **Article.java**
   - `incrementCommentCount()`: 댓글 수 증가 메서드 추가
   - `decrementCommentCount()`: 댓글 수 감소 메서드 추가 (음수 방지 조건 포함)

2. **CommentServiceImpl.java**
   - `addComment()`: 댓글 생성 시 `article.incrementCommentCount()` 호출
   - `removeCommentLogical()`: 댓글 삭제 시 `comment.getArticle().decrementCommentCount()` 호출

### 기술적 특징
- JPA dirty checking 패턴 활용 (명시적 save() 호출 불필요)
- @Transactional 범위 내에서 카운트 업데이트 보장
- decrementCommentCount()에서 음수 방지 로직 구현

## 📷 Screenshot

<img width="2062" height="1314" alt="image" src="https://github.com/user-attachments/assets/8ac2bc5c-5d45-4d79-996e-77cdc034e055" />
<img width="1548" height="440" alt="image" src="https://github.com/user-attachments/assets/b20e6bfa-0c2f-4f59-b35c-2313489e78d4" />

### 삭제 시

<img width="1900" height="1172" alt="image" src="https://github.com/user-attachments/assets/fd4817aa-056d-407a-a187-726400534576" />

<img width="2178" height="846" alt="image" src="https://github.com/user-attachments/assets/f470aeff-e54a-4b23-98fb-6f71c154bea3" />


## 📚 Reference

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## 릴리스 노트

* **버그 수정**
  * 댓글 추가/삭제 시 게시물의 댓글 수가 정확하게 갱신되도록 개선했습니다.
  * 댓글 수가 음수로 내려가지 않도록 보호 로직을 추가해 안정성을 높였습니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yuhandemian committed:
feat: 댓글 생성/삭제 시 게시글 댓글 수 자동 업데이트 기능 추가

Namsoo315 commented:
이것도 마찬가지로 test 통과안한 이유가 serviceImpl에 increment추가가 되서 테스트에도 적용시켜야 해서 그럴겁니다.
제가 하겠습니다.

yuhandemian committed:
fix: 댓글 저장 실패 시 commentCount 불일치 방지

- commentRepository.save() 성공 후 article.incrementCommentCount() 호출
- 댓글 저장 실패 시 카운트가 증가되지 않아 데이터 일관성 보장

yuhandemian committed:
test: removeCommentLogical 테스트 mock 수정

- existsById에서 findById로 변경하여 실제 구현과 일치
- Comment 객체 mock 추가로 decrementCommentCount 호출 지원

Namsoo315 reviewed (approved):
고생하셨습니다.
